Tasks/AzureIoTEdgeV2/deployimage.ts

```diff
@@ -111,6 +110,32 @@ class azureclitask {
   }
 
   static installAzureCliIotExtension(installCommand: string[]) {
+    let outputStream: EchoStream = new EchoStream();
+    let execOptions: IExecOptions = {
+      errStream: outputStream as stream.Writable
+    } as IExecOptions;
+    
+    // check azcli version
+    let checkAzureIoTVersionExtensionCommand = ["--version"]; 
+    let viewAzVersionResult = tl.execSync('az', checkAzureIoTVersionExtensionCommand, execOptions);
+    if(viewAzVersionResult.code !== 0)
+    {
+      throw new Error(`View Az Version Error: ${outputStream.content}`);
+    }
+
+    // Upgrade setuptools in azcli's private python environment to avoid conflict when installing azure-iot extension
+    // temporary solution until the paho-mqtt error in win2016 goes away 
+    if(tl.osType() === Constants.osTypeWindows)
+    {
+      let pythonExeLocation : string = 'C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI2\\python.exe';
+      let setupToolsInstallationCommand = ['-m', 'pip', 'install', '-U', 'setuptools==52.0.0'];
+      let upgradeSetuptoolsResult = tl.execSync(pythonExeLocation, setupToolsInstallationCommand, execOptions);
```

Comment:
It would be good to consider conditionally installing `setuptools` if its not already installed in the az cli environment (leveraging `C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI2\\python.exe python -m pip show setuptools`).

Reply:
@cindydeng1998 Since this does not seem like a blockig change, kindly pick this up in the next iteration.